crates/node/builder/src/launch/exex.rs

```diff
@@ -91,15 +92,30 @@ impl<Node: FullNodeComponents + Clone> ExExLauncher<Node> {
 
         // send notifications from the blockchain tree to exex manager
         let mut canon_state_notifications = components.provider().subscribe_to_canonical_state();
+        let mut finalized_block_notifications = components.provider().finalized_block_stream();
         let mut handle = exex_manager_handle.clone();
+
         components.task_executor().spawn_critical(
             "exex manager blockchain tree notifications",
             async move {
-                while let Ok(notification) = canon_state_notifications.recv().await {
-                    handle
-                        .send_async(notification.into())
-                        .await
-                        .expect("blockchain tree notification could not be sent to exex manager");
+                loop {
+                    select! {
+                        Ok(notification) = canon_state_notifications.recv() => {
+                            handle
+                                .send_async(notification.into())
+                                .await
+                                .expect("blockchain tree canonical state notification could not be sent to exex manager");
+                        }
+                        block_header = finalized_block_notifications.next() => {
+                            if let Some(notification) = block_header{
+                                handle
+                                    .send_async(notification.into())
+                                    .await
+                                    .expect("blockchain tree finalized block notification could not be sent to exex manager");
+                            }
+                        }
```

Comment:
> `Poll::Ready(None)` means that the stream has terminated, and poll_next should not be invoked again.

from https://docs.rs/futures-core/0.3.30/futures_core/stream/trait.Stream.html#tymethod.poll_next

```suggestion
                        Some(block_header) = finalized_block_notifications.next() => {
                            handle
                                .send_async(notification.into())
                                .await
                                .expect("blockchain tree finalized block notification could not be sent to exex manager");
                        }
```